source/tyr/tyr/resources.py

```diff
@@ -1634,3 +1636,43 @@ def put(self, instance_name):
             return_status = 404
 
         return {'action': return_msg}, return_status
+
+
+class Cities(flask_restful.Resource):
+    def check_db(self):
+        cities_db = sqlalchemy.create_engine(current_app.config['CITIES_DATABASE_URI'])
+        try:
+            cities_db.connect()
+            result = cities_db.execute("SELECT version_num FROM alembic_version")
+            for row in result:
+                return row['version_num']
+        except Exception as e:
+            logging.exception("cities db not created : {}".format(e.message))
+
+    def get(self):
+        msg = self.check_db()
+        if msg:
+            return {'message': 'cities db alembic version = {}'.format(msg)}, 200
+        else:
+            return {'message': 'cities db not reachable'}, 404
+
+    def post(self):
+        if not self.check_db():
+            return {'message': 'cities db not reachable'}, 404
+        else:
+            parser = reqparse.RequestParser()
+            parser.add_argument('file', type=werkzeug.FileStorage, location='files')
+            args = parser.parse_args()
+
+            if not args['file']:
+                logging.info("No file provided")
+                return {'message': 'No file provided'}, 400
+
+            osm_file = args['file']
+            osm_file_path = str(os.path.join(os.path.abspath(current_app.config['CITIES_OSM_FILE_PATH']), "file"))
+            osm_file.save(osm_file_path)
+
+            if cities.delay(osm_file_path) != 0:
+                return {'message': 'cities failed!'}, 400
+
+            return {'message': 'OK'}, 200
```

Comment:
Not sure either: As cities import can be long, sender's side should timeout before getting that message.
Shouldn't we manage it like import_data and use finish_job? That we sender can check tyr's /jobs endpoint and see if it's job is failed, pending or done...
Letting @kinnou02 react on that, but as it does the job, it's at least a good first step to me.

Reply:
[`delay()`](http://docs.celeryproject.org/en/latest/reference/celery.app.task.html#celery.app.task.Task.delay) return a future, so it won't wait if  we don't call `wait()`, bu it's a bit strange that comparing it to `0` doesn't fail

Reply:
the comparison to 0 was put when the task was called without the delay. It was a mistake not to remove it when the delay() was used. However, you're right, I don't get how it doesn't fail because the id of the celery AsyncResult returned isn't 0.
???